### PR TITLE
Adds `ItemUse` to `sway-fmt-v2`

### LINKED ISSUE #1808: Formatter incorrectly reformatting use declarations that exceed line length limit
before reformatting 
``` rust
use std::{address::Address, assert::assert, chain::auth::*, context::*, context::call_frames::*, token::*};
```

after reformatting: 
``` rust
stduse:: {
    address::Address, assert::assert, chain::auth:: * , context:: * , context::call_frames:: * , token:: * 
};
```

This then fails to build. 

setup:
``` sh
forc 0.14.5
fuel-core v0.8.0
```

Editor: VSCode

### LINKED ISSUE #2298: Add `ItemUse` to `sway-fmt-v2`

### PR DESCRIPTION
**Pull request by eureka-cpu**
Opening this up for visibility.

- [x] #2226 <-- I've added sorting but only in the scope of each import, not the collective of imports
- [x] Closes #1808
- [x] #2299 #2390

Closes #2298

### CODE REVIEW

File: sway-fmt-v2/src/items/item_use/tests.rs
Comment (mitchmindtree) [inline sway-fmt-v2/src/items/item_use/tests.rs:80]: It could be nice to have a brief comment on these macros explaining what's happening, just to give future readers/maintainers a head-start.

File: sway-fmt-v2/src/items/item_use/tests.rs
Comment (mitchmindtree) [inline sway-fmt-v2/src/items/item_use/tests.rs:74]: Do we do this diff printing with different colours in other places? If so, maybe we can move this to a `print_diff` function or similar? If not, we can probably wait until we do this in multiple places before we refactor it out.

**Review by mitchmindtree — APPROVED** [review]

**Comment by kayagokalp** [comment]
Merging master (as #2518 merged) here to resolve CI failure

File: sway-fmt-v2/src/items/item_use/tests.rs
Comment (eureka-cpu) [reply sway-fmt-v2/src/items/item_use/tests.rs:80]: Since it's a copypasta I'll go ahead an update both in a separate PR

File: sway-fmt-v2/src/items/item_use/tests.rs
Comment (eureka-cpu) [reply sway-fmt-v2/src/items/item_use/tests.rs:74]: It is in one other place, `expr`, but I can do that in another PR as well. Perhaps the same PR for adding comments describing the macros

**Review by kayagokalp — APPROVED** [review]
Tested locally, works & looks perfect!

Maybe something that we can address with the later PR you mentioned in @mitchmindtree's review: After this PR merges we only have `item_use` folder (not counting `expr` as it resides in utils folder) which holds the tests for formatting. For consistency maybe we can either have this kind of structure for all of the items and removing the corresponding tests from `fmt.rs`, or move these tests to `fmt.rs` (possibly expending/reusing your cool macro for other tests)

**Pull request merged**
